pkg/flowcli/services/transactions.go

```diff
@@ -53,87 +50,159 @@ func NewTransactions(
 	}
 }
 
-// Send transaction
-func (t *Transactions) Send(
-	transactionFilename string,
-	signerName string,
-	args []string,
-	argsJSON string,
+// GetStatus of transaction
+func (t *Transactions) GetStatus(
+	transactionID string,
+	waitSeal bool,
 ) (*flow.Transaction, *flow.TransactionResult, error) {
-	if t.project == nil {
-		return nil, nil, fmt.Errorf("missing configuration, initialize it: flow project init")
-	}
+	txID := flow.HexToID(
+		strings.ReplaceAll(transactionID, "0x", ""),
+	)
 
-	signer := t.project.AccountByName(signerName)
-	if signer == nil {
-		return nil, nil, fmt.Errorf("signer account: [%s] doesn't exists in configuration", signerName)
-	}
+	t.logger.StartProgress("Fetching Transaction...")
 
-	code, err := util.LoadFile(transactionFilename)
+	tx, err := t.gateway.GetTransaction(txID)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	return t.send(code, signer, args, argsJSON)
+	if waitSeal {
+		t.logger.StartProgress("Waiting for transaction to be sealed...")
+	}
+
+	result, err := t.gateway.GetTransactionResult(tx, waitSeal)
+
+	t.logger.StopProgress()
+
+	return tx, result, err
 }
 
-// SendForAddressWithCode send transaction for address and private key specified with code
-func (t *Transactions) SendForAddressWithCode(
-	code []byte,
-	signerAddress string,
-	signerPrivateKey string,
+// Sign transaction
+func (t *Transactions) Sign(
+	signerName string,
+	proposerName string,
+	payerAddress string,
+	additionalAuthorizers []string,
+	role string,
+	scriptFilename string,
+	payloadFilename string,
 	args []string,
 	argsJSON string,
-) (*flow.Transaction, *flow.TransactionResult, error) {
-	address := flow.HexToAddress(signerAddress)
+	approveSigning bool,
+) (*project.Transaction, error) {
+	if t.project == nil {
+		return nil, fmt.Errorf("missing configuration, initialize it: flow project init")
+	}
 
-	privateKey, err := crypto.DecodePrivateKeyHex(crypto.ECDSA_P256, signerPrivateKey)
-	if err != nil {
-		return nil, nil, fmt.Errorf("private key is not correct")
+	// get the signer account
+	signerAccount := t.project.AccountByName(signerName)
+	if signerAccount == nil {
+		return nil, fmt.Errorf("signer account: [%s] doesn't exists in configuration", signerName)
 	}
 
-	account := project.AccountFromAddressAndKey(address, privateKey)
+	if payloadFilename != "" && scriptFilename != "" {
+		return nil, fmt.Errorf("can not use both a transaction payload and Cadence code file")
+	} else if payloadFilename == "" && scriptFilename == "" {
+		return nil, fmt.Errorf("provide either a transaction payload or Cadence code file")
+	}
```

Comment:
Am I correct in assuming that most of the other fields are not required if `payload` is passed? 

For example, the `proposer` arg would be ignored in this case, right?

```
flow transactions sign --payload=mypayload.rlp --proposer=0xabcdef123
```

Do you think it would make sense to throw an error in this case?

Reply:
You sir are correct and yes, fixed :)